libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Functions/IsDialogActiveFunction.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using AdaptiveExpressions;
+using AdaptiveExpressions.Memory;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Dialogs.Functions
+{
+    /// <summary>
+    /// Defines isDialogActive(id) expression function.
+    /// </summary>
+    /// <remarks>
+    /// This expression will return true if any of the dialogIds is on the dialog execution stack.
+    /// You can pass 1..N dialog ids.
+    /// </remarks>
+    /// <example>
+    /// isDialogActive('dialog1')
+    /// isDialogActive('dialog1', 'dialog2', 'dialog3').
+    /// </example>
+    public class IsDialogActiveFunction : ExpressionEvaluator
+    {
+        public const string Name = "isDialogActive";
+        public const string Alias = "isActionActive";
+
+        public IsDialogActiveFunction()
+            : base(Name, Function, ReturnType.Boolean)
+        {
+        }
+
+        private static (object value, string error) Function(Expression expression, IMemory state, Options options)
+        {
+            if (state.TryGetValue("dialogcontext.stack", out object stackArray))
```

Comment:
Should this be 

```suggestion
            if (state.TryGetValue($"{ScopePath.DialogContext}.{DialogContextMemoryScope.Stack}", out object stackArray))
```

I worry about these being hard to change or track. Defining the constants might help? Thoughts? (note that I assumed the existence of a constant for "stack" also.